src/api/file.ts

```diff
@@ -0,0 +1,27 @@
+import { useMutation } from "@tanstack/react-query";
+import { FirebaseError } from "firebase/app";
+import {
+  StorageReference,
+  getDownloadURL,
+  getStorage,
+  ref,
+  uploadBytes,
+} from "firebase/storage";
+import { uniqueId } from "../utils/recipeUtils";
+
+const uploadAndGetUrl = async (storageRef: StorageReference, image: File) => {
+    await uploadBytes(storageRef, image)
```

Comment:
what does this function return? 🤔 
I'd hope that if upload succeeded it resolves to object with downloadURL within.

Reply:
It returns only the download URL. 
I honestly have no idea how to do it in a proper way :"(